events_mgmt/src/main/java/com/adobe/event/management/EventMetadataServiceImpl.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2017 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+package com.adobe.event.management;
+
+import static com.adobe.util.Constants.API_MANAGEMENT_URL;
+
+import com.adobe.Workspace;
+import com.adobe.event.management.api.EventMetadataApi;
+import com.adobe.event.management.model.EventMetadata;
+import com.adobe.event.management.model.EventMetadataCollection;
+import com.adobe.util.FeignUtil;
+import feign.RequestInterceptor;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.apache.commons.lang3.StringUtils;
+
+class EventMetadataServiceImpl implements EventMetadataService {
+
+  private final EventMetadataApi eventMetadataApi;
+  private final Workspace workspace;
+
+  EventMetadataServiceImpl(final RequestInterceptor authInterceptor,
+      final Workspace workspace, final String url) {
+    String apiUrl = StringUtils.isEmpty(url) ? API_MANAGEMENT_URL : url;
+    if (authInterceptor == null) {
+      throw new IllegalArgumentException(
+          "EventMetadataService is missing a authentication interceptor");
+    }
+    if (workspace == null) {
+      throw new IllegalArgumentException("EventMetadataService is missing a workspace context");
+    }
+    workspace.validateWorkspaceContext();
+    this.eventMetadataApi = FeignUtil.getDefaultBuilder()
+        .requestInterceptor(authInterceptor)
+        .target(EventMetadataApi.class, apiUrl);
+    this.workspace = workspace;
+  }
+
+  @Override
+  public List<EventMetadata> getEventMetadata(String providerId) {
+    Optional<EventMetadataCollection> eventMetadataCollection =
+        eventMetadataApi.findByProviderId(providerId);
+    if (eventMetadataCollection.isPresent()) {
+      return eventMetadataCollection.get().getEventMetadata();
+    } else {
+      return new ArrayList<>();
+    }
```

Comment:
1. Why not return `Optional<List<EventMetadata>>` here like we have done in other getters?
2. How do we differentiate the cases where "provider id is invalid" and "provider id is valid but there is no event metadata configured"?

Reply:
when the providerId is invalid, this will face a `404` translated into a `FeignException$BadRequest` 
note we could add cleaner error handling as part of https://github.com/adobe/aio-lib-java/issues/7

Reply:
1. I think the interface needs some work. On what conditions do we get empty-optional vs exception?
2. If all 4xx responses are translated to `FeignException`, then why do we need `Optional<T>` as the return type? Shouldn't it simply return `T` then?

Reply:
we get empty Optional when `404`

Reply:
Got it. Suggest documenting this contract in the interface methods. We can document few exception scenarios, cases where an empty-value is returned and where valid non-empty value is returned.

Reply:
done